internal/engine/selectors/selectors.go

```diff
@@ -107,6 +107,7 @@ func newEnvForEntity(varName string, typ any, typName string) (*cel.Env, error)
 }
 
 type compiledSelector struct {
+	orig    string
```

Comment:
there is ast.ToString but it doesn't guarantee that the result will be exactly the same, only that it will have the same meaning..